src/runtime-rs/crates/hypervisor/src/remote/inner.rs

```diff
@@ -0,0 +1,330 @@
+use crate::{
+    device::DeviceType, hypervisor_persist::HypervisorState,
+    remote::protocols::hypervisor::CreateVMRequest, HypervisorConfig, HYPERVISOR_REMOTE,
+};
+use crate::{MemoryConfig, VcpuThreadIds};
+use anyhow::Result;
+use async_trait::async_trait;
+use kata_types::capabilities::{Capabilities, CapabilityBits};
+use persist::sandbox_persist::Persist;
+use std::path::Path;
+use std::{collections::HashMap, time};
+use std::{fs, u64};
+use tokio::sync::{mpsc, Mutex};
+use ttrpc::context::{self};
+use ttrpc::r#async::Client;
+
+use super::protocols::{
+    hypervisor::{StartVMRequest, StopVMRequest},
+    hypervisor_ttrpc::HypervisorClient,
+};
+
+const REMOTE_SCHEME: &str = "remote";
+const DEFAULT_MIN_TIMEOUT: i32 = time::Duration::from_secs(60).as_millis() as i32;
+
+#[derive(Debug)]
+pub struct RemoteInner {
+    /// sandbox id
+    pub(crate) id: String,
+    /// hypervisor config
+    pub(crate) config: HypervisorConfig,
+    /// agent socket path
+    pub(crate) agent_socket_path: String,
+    /// sandbox annotations
+    pub(crate) annotations: HashMap<String, String>,
+    /// netns path
+    pub(crate) netns: Option<String>,
+
+    exit_notify: Option<mpsc::Sender<i32>>,
+    exit_waiter: Mutex<(mpsc::Receiver<i32>, i32)>,
+}
+
+impl RemoteInner {
+    pub fn new() -> Self {
+        let (exit_notify, exit_waiter) = mpsc::channel(1);
+
+        Self {
+            id: "".to_string(),
+            config: HypervisorConfig::default(),
+            agent_socket_path: "".to_string(),
+            annotations: HashMap::new(),
+            netns: None,
+
+            exit_notify: Some(exit_notify),
+            exit_waiter: Mutex::new((exit_waiter, 0)),
+        }
+    }
+
+    pub(crate) async fn prepare_vm(&mut self, id: &str, netns: Option<String>) -> Result<()> {
+        info!(sl!(), "Preparing REMOTE VM");
+        self.id = id.to_string();
+        let c = Client::connect(&format!("unix://{}", &self.config.remote_hypervisor_socket))?;
+        let client = HypervisorClient::new(c.clone());
+
+        if let Some(netns_path) = &netns {
+            debug!(sl!(), "set netns for vmm master {:?}", &netns_path);
+            let netns_fd = std::fs::File::open(netns_path);
+            if netns_fd.is_err() {
+                error!(sl!(), "failed to open netns file {:?}", netns_path);
+            } else {
+                info!(sl!(), "set netns for vmm master {:?}", &netns_path);
+            }
+        }
+
+        let ctx = context::with_timeout(10_0000_0000);
```

Comment:
Comments needed to clarify the reason of using `10_0000_0000`

Reply:
The timeout have changed to the default value. Makes it the same action as Go.